Add chosenByUser support for LFTrack / LFScrobbleRequest

tardisx opened the pull request:
A simple patch to support chosenByUser for scrobbles. Details at:

http://www.last.fm/api/scrobbling#when-to-set-the-chosenbyuser-parameter

tardisx committed:
Added chosenByUser boolean field for tracks, defaults to true.

13bold commented:
I think we would indeed like to support this feature, but with a few small tweaks to the patch you provided. Predominately, we need to find an elegant way to make sure that `chosenByUser` is only sent if _explicitly_ set by the user, since the Last.fm docs state that it should simply be omitted if there is any ambiguity.

tardisx commented:
It probably doesn't matter (at least, I'd hope not), but agreed. I'll make that change.

tardisx committed:
Only supply the chosenByUser parameter when appropriate.

According to the last.fm api reference, this parameter should only
be supplied when the track was not 'user chosen' and then set to 0.
Any other time it should not be supplied at all.
